plugins/experimental/icap/icap_plugin.cc

```diff
@@ -0,0 +1,1267 @@
+/** @file
+
+  @brief A plugin that sends the http body through an ICAP request to a
+         scanner server. If malicious content is detected, then the scanner
+         will return an error message body, which we will pass to the user
+         agent. Otherwise it will return the same content that was passed to
+         it, in which case we will pass the content to user agent.
+
+  @section license License
+
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+#include <string>
+#include <regex>
+
+#include <netinet/in.h>
+#include <arpa/inet.h>
+#include <iostream>
+#include <sstream>
+
+#include "ts/ts.h"
+
```

Comment:
Everything after this, except for TSPluginInit() can go in an anonymous namespace, eliminating the need for "static".